testsuite/src/Test/Tasty/Clash.hs

```diff
@@ -269,15 +273,17 @@ instance IsTest ClashTest where
         (testExitCode testExit) "clash" (args oDir) NoGlob PrintNeither False
         (specificExitCode testExit) (ExpectStdErr expectedErr) Nothing
 
+    modFlags = if ctLoadFromInterface
+                 then [ctModName]
+                 else ["-i" <> ctSourceDirectory
+                      , ctSourceDirectory </> ctModName <.> "hs"]
+
     args oDir =
-      [ target
-      , ctSourceDirectory </> ctModName <.> "hs"
-      , "-i" ++ ctSourceDirectory
-      , "-fclash-hdldir", oDir
+      target : modFlags <>
+      [ "-fclash-hdldir", oDir
       , "-odir", oDir
       , "-hidir", oDir
       , "-fclash-debug", "DebugSilent"
-      , "-package", "clash-testsuite"
```

Comment:
Could you see if Stack needs this line by any chance? So something like `stack run testsuite -- -p T1796`.